plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -270,6 +280,70 @@ private void testAddNestedColumnWithColumnMappingMode(String columnMappingMode)
                 .containsPattern("(delta\\.columnMapping\\.physicalName.*?){11}");
     }
 
+    @Test // regression test for https://github.com/trinodb/trino/issues/24121
+    void testPartitionValuesParsedCheckpoint()
+            throws Exception
+    {
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.ID);
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.NAME);
+        testPartitionValuesParsedCheckpoint(ColumnMappingMode.NONE);
+    }
+
+    private void testPartitionValuesParsedCheckpoint(ColumnMappingMode columnMappingMode)
+            throws Exception
+    {
+        try (TestTable table = new TestTable(
+                getQueryRunner()::execute,
+                "test_checkpoint",
+                "(x int, part int) WITH (checkpoint_interval = 3, column_mapping_mode = '" + columnMappingMode + "', partitioned_by = ARRAY['part'])")) {
```

Comment:
Can you add also test for other types (like Date) which has different representation in `PartitionValues` and `PartitionValues_parsed`